tests/keras/backend/reference_operations.py

```diff
@@ -126,8 +126,13 @@ def pool(x, pool_size, strides, padding, data_format, pool_mode):
 
 
 def rnn(x, w, init, go_backwards=False, mask=None, unroll=False, input_length=None):
-    w_i, w_h, w_o = w
+    # Test these recurrent equations:
+    # h_t = w_o( x_{t} w_i + h_{t-1} w_h + s_{t-1} w_s )
```

Comment:
Thank you for the PR, @PeterChe1990. Are these canonical operations for RNN? Is there any reference article?

Reply:
This is just an extension of the previous rnn function with an additional hidden state `s`. More complicated states are used in other LSTM/GRU-like RNNs, but the example here is simple to make the test succinct.